### PR TITLE
new file format -- old format refactoring

### PR DESCRIPTION
**Pull request by abuts**
This is the first part of the ticket, modifying existing file format and extracting the part, common with new file format. 
Almost no new file format is defined -- just empty stub for it

### CODE REVIEW

File: horace_core/sqw/file_io/@horace_binfile_interface/build_app_header.m [hunk 1]
```diff
@@ -0,0 +1,47 @@
+function   app_header = build_app_header(obj,obj_to_save)
+% Build Horace sqw-file header to write to hdd. The header contains
+% information on type of sqw object and sqw-file subversion
```
Comment (cmarooney-stfc) [inline horace_core/sqw/file_io/@horace_binfile_interface/build_app_header.m:3]: what does subversion mean here?

File: horace_core/sqw/file_io/@horace_binfile_interface/build_app_header.m [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+%           throw 'HORACE:sqw_file_interface:invalid_argument' exception.
+%
+
+if ~exist('obj_to_save','var')
+    obj_to_save = obj.sqw_holder_;
+end
+%
+if isempty(obj_to_save)
+    error('HORACE:sqw_file_interface:invalid_argument',...
+        'Trying to build_app_headed but no object to build header from is provided')
```
Comment (cmarooney-stfc) [inline horace_core/sqw/file_io/@horace_binfile_interface/build_app_header.m:24]:
```suggestion
        'Trying to build_app_header but no object to build header from is provided')
```

File: horace_core/sqw/file_io/@binfile_v2_common/define_upgrade_map_.m [hunk 3]
```diff
@@ -0,0 +1,61 @@
+function obj=define_upgrade_map_(obj,file_exist,old_ldr)
```
Comment (cmarooney-stfc) [inline horace_core/sqw/file_io/@binfile_v2_common/define_upgrade_map_.m:1]: be good to have a description of what this method does, also for arguments

File: horace_core/sqw/@SQWDnDBase/SQWDnDBase.m [hunk 4]
```diff
@@ -163,6 +163,5 @@
         status = adjust_aspect(w);
         [ok,mess,adjust,present]=adjust_aspect_option(args_in);
         dout = smooth_dnd(din, xunit, varargin);
-    end
+    end
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@SQWDnDBase/SQWDnDBase.m:166]: looks like unncessary spaces added - accident? better to withdraw this commit?

File: admin/convert_uoffset_to_crystal_cartesian.m [hunk 5]
```diff
@@ -1,6 +1,6 @@
 function convert_uoffset_to_crystal_cartesian(filename,varargin)
 % Read uoffset, currently stored in sqw files, and convert this offset into
-% crystal cartesian system coordinat if the value, specified in the sqw
+% crystal Cartesian system coordinate if the value, specified in the sqw
```
Comment (cmarooney-stfc) [inline admin/convert_uoffset_to_crystal_cartesian.m:3]:
```suggestion
% crystal Cartesian system coordinates if the value, specified in the sqw
```
Following this change it says "if..." but it the sentence is incomplete and does not say what this condition is

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/file_io/CollabDiagram.gv [hunk 6]
```diff
@@ -1,10 +1,10 @@
 	/*edge [dir=back,arrowhead=odiamond];*/digraph CollabDiag {
    /* Collaboration&inheritance diagram showing relationship between various classes supporting sqw file access*/
 	edge [dir=back];
-	dnd_binfile_common [shape=box];
+	horace_binfile_interface [shape=box];
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/CollabDiagram.gv:4]: we need to have `graphvis`, but I will generate inheritance diagram from this.

File: horace_core/sqw/file_io/@sqw_serializer/sqw_serializer.m [hunk 7]
```diff
@@ -4,15 +4,20 @@
...
     properties(Access=private,Hidden=true)
         sqw_holder_ = []; % reference to sqw object to serialize (if any)
         n_header_ = 1; % number of header to process
         %
-        base_classes_ = {'double','single','int8','uint8','int16','uint16',...
-            'int32','uint32','int64','uint64','float64'};
-        class_sizes_ =  [8,4,1,1,2,2,4,4,8,8,8]; % in bytes
-        class_map_; % map to associate class name and class size
-        
+
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/@sqw_serializer/sqw_serializer.m:20]: done

File: horace_core/sqw/file_io/@horace_binfile_interface/set_file_to_update.m [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+% Set filename to save sqw data and open file for write/append/update
+% operations
+%
+% Usage
+% >> obj = obj.set_file_to_update(); % reopen existing file for
+%          write/update operations
+%
+% >> obj = obj.set_file_to_update(new_filename);
+%
+% Open new or existing sqw file to perform write/append operations
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/@horace_binfile_interface/set_file_to_update.m:11]: hopefully fixed, though not sure fully understood the remark. It is description, and then two slightly different usage scenario for two input types

File: horace_core/sqw/file_io/@horace_binfile_interface/set_file_to_update.m [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+%
+% Ouptputs:
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/@horace_binfile_interface/set_file_to_update.m:13]: done

File: horace_core/sqw/file_io/@horace_binfile_interface/set_file_to_update.m [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+% operations
+%
+% Usage
+% >> obj = obj.set_file_to_update(); % reopen existing file for
+%          write/update operations
+%
+% >> obj = obj.set_file_to_update(new_filename);
+%
+% Open new or existing sqw file to perform write/append operations
+%
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/@horace_binfile_interface/set_file_to_update.m:12]: hopefully fixed as above

File: horace_core/sqw/file_io/@horace_binfile_interface/is_activated.m [hunk 11]
```diff
@@ -0,0 +1,29 @@
+function is = is_activated(obj, read_or_write)
+% Check if the file-accessor is bound to an open binary file
+%
+% Input
+% -----
+%
+% read_or_write   Char array. If 'read' return true if file is open
+%                 for reading. If 'write' return true if file is
+%                 open for writing.
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/@horace_binfile_interface/is_activated.m:9]: added

File: horace_core/sqw/file_io/@horace_binfile_interface/get_file_header.m [hunk 12]
```diff
@@ -12,29 +12,38 @@
...
 %
-%
-%
-%
-[header,fid,message] = read_header_(file,varargin{:});
-if ~isempty(message)
+
+isnum = cellfun(@(x)isnumeric(x),varargin);
+if any(isnum)
+    max_buffer_size = varargin{isnum};
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/@horace_binfile_interface/get_file_header.m:23]: the form used for processing ignores all other possible numeric values except the first one.  This is how it should work as only one numeric value (buffer size) is processed.

File: horace_core/sqw/file_io/@horace_binfile_interface/build_app_header.m (hunk 1 above)
Comment (abuts) [reply horace_core/sqw/file_io/@horace_binfile_interface/build_app_header.m:3]: hopefully clarified.

File: horace_core/sqw/file_io/@horace_binfile_interface/build_app_header.m (hunk 2 above)
Comment (abuts) [reply horace_core/sqw/file_io/@horace_binfile_interface/build_app_header.m:24]: modified

File: horace_core/sqw/@SQWDnDBase/SQWDnDBase.m (hunk 4 above)
Comment (abuts) [reply horace_core/sqw/@SQWDnDBase/SQWDnDBase.m:166]: removed

File: horace_core/sqw/file_io/@binfile_v2_common/define_upgrade_map_.m (hunk 3 above)
Comment (abuts) [reply horace_core/sqw/file_io/@binfile_v2_common/define_upgrade_map_.m:1]: added

File: horace_core/utilities/update_pixels_run_id.m [hunk 13]
```diff
@@ -3,7 +3,7 @@
 % in headers may or may not be consistent with the run-id(s) stored in
 % pixels.
 %
-% The routine tries to analyze pixels run_id(s) and check if the id-s are
+% The routine tries to analyse pixels run_id(s) and check if the id-s are
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/utilities/update_pixels_run_id.m:6]: I am relying on notepad++ speller to notice such issues. If a word is marked red and suggestion appears, I do change it to reduce red lines in the text.

**Pull request merged**